### PR TITLE
Bratseth/system info

### PR DESCRIPTION
**Pull request by bratseth**
@hmusum AND @gjoranv please review

### CODE REVIEW

**Review by hmusum — APPROVED** [review]
👍

**Review by hmusum — CHANGES_REQUESTED** [review]
Fails abi check:
[INFO] --- abi-check-plugin:7-SNAPSHOT:abicheck (default) @ container-core ---
[ERROR] Extra class: ai.vespa.cloud.Environment
[ERROR] Extra class: ai.vespa.cloud.SystemInfo
[ERROR] Extra class: ai.vespa.cloud.Zone

File: container-core/src/main/java/ai/vespa/cloud/SystemInfo.java [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+ * This is available and can be injected when running in a cloud environment.
+ *
+ * @author bratseth
+ */
 public class SystemInfo {
+
+    private final Zone zone;
+
+    /** Do not use */
+    @SuppressWarnings("unused")
```
Comment (gjoranv) [inline container-core/src/main/java/ai/vespa/cloud/SystemInfo.java]: Suppressing "unused" should be unnecessary when you have the Inject annotation.

File: container-core/src/main/java/ai/vespa/cloud/SystemInfo.java [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+ * Provides information about the system in which this container is running.
+ * This is available and can be injected when running in a cloud environment.
+ *
+ * @author bratseth
+ */
 public class SystemInfo {
+
+    private final Zone zone;
+
+    /** Do not use */
```
Comment (gjoranv) [inline container-core/src/main/java/ai/vespa/cloud/SystemInfo.java]: I suggest to skip this comment. What does it mean?

File: container-core/src/main/java/ai/vespa/cloud/Zone.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        this.environment = environment;
+        this.region = region;
+    }
+
+    public Environment environment() { return environment; }
+    public String region() { return region; }
+
+    /** Returns the string environment.region */
+    @Override
+    public String toString() { return environment + "." + region; }
```
Comment (gjoranv) [inline container-core/src/main/java/ai/vespa/cloud/Zone.java]: Perhaps this should be e.g. ´stringValue()´, and return a String that is less tempting to parse from toString.

**Review by gjoranv — APPROVED** [review]

File: container-core/src/main/java/ai/vespa/cloud/SystemInfo.java (hunk 1 above)
Comment (bratseth) [reply container-core/src/main/java/ai/vespa/cloud/SystemInfo.java]: IntelliJ don't think so, but I guess it's clear enough with the annotation anyway.

File: container-core/src/main/java/ai/vespa/cloud/SystemInfo.java (hunk 2 above)
Comment (bratseth) [reply container-core/src/main/java/ai/vespa/cloud/SystemInfo.java]: It means to not use it ... I probably want to switch to another config later which will break any who use this method.

File: container-core/src/main/java/ai/vespa/cloud/Zone.java (hunk 3 above)
Comment (bratseth) [reply container-core/src/main/java/ai/vespa/cloud/Zone.java]: I would agree if this was further in but I think this is fine for an "edge API" class.

**Pull request merged**